assets/src/edit-story/components/panels/panel/content.js

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { useContext } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import panelContext from './context';
+
+const Form = styled.form`
+	margin: 10px 20px;
+	overflow: auto;
+
+	${ ( { hidden } ) => hidden && 'display: none' }
+`;
+
+function Content( { children, onSubmit, ...rest } ) {
+	const { state: { isCollapsed, height, panelContentId } } = useContext( panelContext );
+
+	const formStyle = {
+		height: height === null ? 'auto' : `${ height }px`,
+	};
+
+	return (
+		<Form
+			style={ formStyle }
+			onSubmit={ onSubmit }
+			{ ...rest }
+			id={ panelContentId }
+			hidden={ isCollapsed }
+		>
+			{ children }
+		</Form>
+	);
+}
+
+Content.propTypes = {
+	children: PropTypes.oneOfType( [
+		PropTypes.arrayOf( PropTypes.node ),
+		PropTypes.node,
+	] ).isRequired,
+	onSubmit: PropTypes.func,
+};
+
+Content.defaultProps = {
+	onSubmit: ( evt ) => evt.preventDefault(),
```

Comment:
@dvoytenko in response to your comment https://github.com/ampproject/amp-wp/pull/4138#discussion_r372116705:

I've defaulted the `onSubmit` parameter to an otherwise empty `preventDefault()`er as suggested. It makes good sense to do.

Regarding your question whether we should have a form or not: Placing an input inside a form has several advantages and inputs should only be "orphaned" under very special circumstances.

1.  It's semantically correct. This is a form and the inputs' relation is evident by them being in the same form.
2. Screen readers treat form contents in a special way optimized for form element interaction. Users of screen readers expect inputs to be grouped in forms.
3. It allows us to do submit handling on keypress enter inside input fields without adding a handler to the individual input field (as enter in an input field in a form by default submits the form).

Reply:
Sounds good. Let's resolve.